### PR TITLE
Allow deletion of entries by label

### PR DESCRIPTION
**Pull request by bdk0**
This command will delete any entry with the label 'Debian':

efibootmgr -B -L Debian 

If more than one entry have the same label, all will be deleted

### CODE REVIEW

File: src/efibootmgr.c [hunk 1]
```diff
@@ -621,6 +621,38 @@ delete_var(const char *prefix, uint16_t num)
 	return 0;
 }
 
+static int
+delete_label(const char *prefix, const unsigned char *label)
+{
+	list_t *pos;
+	var_entry_t *boot;
+	int num_deleted=0;
```
Comment (martinezjavier) [inline src/efibootmgr.c:629]: `int num_deleted = 0;`

File: src/efibootmgr.c [hunk 2]
```diff
@@ -621,6 +621,38 @@ delete_var(const char *prefix, uint16_t num)
...
+	int rc;
+	efi_load_option *load_option;
+	const unsigned char *desc;
+
+	list_for_each(pos, &entry_list) {
+		boot = list_entry(pos, var_entry_t, list);
+		load_option = (efi_load_option *)boot->data;
+		desc = efi_loadopt_desc(load_option, boot->data_size);
+
+		if (strcmp((char*)desc,(char*)label)==0) {
```
Comment (martinezjavier) [inline src/efibootmgr.c:639]: here too ` == 0`

File: src/efibootmgr.c [hunk 3]
```diff
@@ -621,6 +621,38 @@ delete_var(const char *prefix, uint16_t num)
...
+			rc = delete_var(prefix,boot->num);
+			if (rc <0) {
```
Comment (martinezjavier) [inline src/efibootmgr.c:641]: `rc < 0`

**Comment by martinezjavier** [comment]
@bdk0 patch looks good to me. Just mentioned a few coding style nits.

Do you think that the man page should be updated to make clear that this functionality is supported? The long version of the `-B` option is `--delete-bootnum`, so unless there's something mentioned in the docs users will assume that only a boot number could be used.

**Comment by vathpela** [comment]
I've pushed this with @martinezjavier 's suggestions and a few other minor coding style changes, plus a more explanatory commit message, as 9af1ab0cecd9dbd123aed008567eb7382706f153 .

**Pull request closed without merging**